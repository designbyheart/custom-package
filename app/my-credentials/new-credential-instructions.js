// @flow

import React, { PureComponent } from 'react'
import {
  ImageBackground,
  Image,
  TouchableOpacity,
  StyleSheet,
  Linking,
  Text,
  View,
} from 'react-native'
import Snackbar from 'react-native-snackbar'

import { scale, verticalScale, moderateScale } from 'react-native-size-matters'

import type { NewCredentialInstructionsProps } from './type-my-credentials'
import { colors, fontFamily } from '../common/styles/constant'

export class NewCredentialInstructions extends PureComponent<
  NewCredentialInstructionsProps,
  void
> {
  render() {
    const { usingProductionNetwork } = this.props
    return (
      <ImageBackground
        source={require('../images/connection-items-placeholder.png')}
        style={styles.backgroundImage}
      >
        <View style={[styles.contentContainer]}>
          <View>
            <Image
              source={require('../images/CredentialGraphic.png')}
              resizeMode="contain"
              style={styles.image}
            />
            <Text style={[styles.headline]}>
              No Credentials yet!
            </Text>
            <Text style={[styles.title]}>Want to see how this app works?</Text>
            {usingProductionNetwork && (
              <Text style={[styles.text]}>
                Go through the tutorial at www.try.connect.me. 
                Connect.Me is for collecting digital Credentials. They will appear here.
              </Text>
            )}
            {!usingProductionNetwork && (
              <Text style={[styles.text]}>
                Connect.Me is for collecting digital Credentials. They will appear here.
              </Text>
            )}
          </View>
        </View>
      </ImageBackground>
    )
  }

  openTryConnectMe = () => {
    Linking.openURL('https://try.connect.me').catch(() => {
      Snackbar.show({
        text:
          'Could not open Tutorial website. No capable browser found to open it.',
        duration: Snackbar.LENGTH_LONG,
        backgroundColor: colors.cmRed,
        textColor: colors.cmWhite,
      })
    })
  }
}

const styles = StyleSheet.create({
  backgroundImage: {
    width: '100%',
    height: '100%',
    alignItems: 'center',
    justifyContent: 'center',
  },
  contentContainer: {
    width: '100%',
    height: '100%',
    alignItems: 'center',
    justifyContent: 'center',
  },
  image: {
    height: moderateScale(90, 2.5),
    alignSelf: 'center',
  },
  headline: {
    fontSize: moderateScale(20, 0.1),
    color: colors.cmOrange,
    marginVertical: moderateScale(5),
    fontFamily: fontFamily,
    textAlign: 'center',
    fontWeight: 'bold',
  },
  title: {
    fontSize: moderateScale(17, 0.1),
    marginVertical: moderateScale(5),
    color: colors.cmGray1,
    textAlign: 'center',
  },
  text: {
    fontSize: moderateScale(15, 0.1),
    marginVertical: moderateScale(5),
    marginHorizontal: '5%',
    color: colors.cmGray1,
    textAlign: 'center',
  },
})
