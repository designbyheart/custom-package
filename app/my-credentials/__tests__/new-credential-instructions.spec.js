// @flow

import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'
import { NewCredentialInstructions } from '../new-credential-instructions'

describe('<NewCredentialInstructions />', () => {
  it('should match snapshot', () => {
    const component = renderer.create(
      <NewCredentialInstructions usingProductionNetwork={true} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should match snapshot for test network', () => {
    const component = renderer.create(
      <NewCredentialInstructions usingProductionNetwork={false} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
