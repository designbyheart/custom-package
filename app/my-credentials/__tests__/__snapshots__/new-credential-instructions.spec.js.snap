// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewCredentialInstructions /> should match snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "alignItems": "center",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "default": "RelativeImageStub",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
  >
    <View>
      <Image
        resizeMode="contain"
        source={
          Object {
            "default": "RelativeImageStub",
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "height": 70.7142857142857,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#EB9B2D",
              "fontFamily": "Lato",
              "fontSize": 19.82857142857143,
              "fontWeight": "bold",
              "marginVertical": 4.785714285714286,
              "textAlign": "center",
            },
          ]
        }
      >
        No Credentials yet!
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#505050",
              "fontSize": 16.854285714285716,
              "marginVertical": 4.785714285714286,
              "textAlign": "center",
            },
          ]
        }
      >
        Want to see how this app works?
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#505050",
              "fontSize": 14.871428571428572,
              "marginHorizontal": "5%",
              "marginVertical": 4.785714285714286,
              "textAlign": "center",
            },
          ]
        }
      >
        Go through the tutorial at www.try.connect.me. Connect.Me is for collecting digital Credentials. They will appear here.
      </Text>
    </View>
  </View>
</View>
`;

exports[`<NewCredentialInstructions /> should match snapshot for test network 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "alignItems": "center",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "default": "RelativeImageStub",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
  >
    <View>
      <Image
        resizeMode="contain"
        source={
          Object {
            "default": "RelativeImageStub",
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "height": 70.7142857142857,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#EB9B2D",
              "fontFamily": "Lato",
              "fontSize": 19.82857142857143,
              "fontWeight": "bold",
              "marginVertical": 4.785714285714286,
              "textAlign": "center",
            },
          ]
        }
      >
        No Credentials yet!
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#505050",
              "fontSize": 16.854285714285716,
              "marginVertical": 4.785714285714286,
              "textAlign": "center",
            },
          ]
        }
      >
        Want to see how this app works?
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#505050",
              "fontSize": 14.871428571428572,
              "marginHorizontal": "5%",
              "marginVertical": 4.785714285714286,
              "textAlign": "center",
            },
          ]
        }
      >
        Connect.Me is for collecting digital Credentials. They will appear here.
      </Text>
    </View>
  </View>
</View>
`;
